### PR TITLE
maintenance/adding-index-page

### PR DESCRIPTION
**Pull request by daviduzumeri**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,27 +1,36 @@
 {
   "name": "@touchbistro/tbui",
-  "version": "0.0.1",
+  "version": "0.0.5",
+  "main": "dist/index.js",
```
Comment (clayton-duarte) [inline package.json:4]: Finally, we have a propper module! Thanks man

File: package.json [hunk 2]
```diff
@@ -1,27 +1,36 @@
...
   "scripts": {
-    "dev": "docz dev",
-    "build": "docz build",
-    "start": "docz build && docz serve",
+    "build": "tsc -p tsconfig.build.json",
+    "dev:docs": "docz dev",
```
Comment (clayton-duarte) [inline package.json:7]: I think we can keep it as just `yarn dev` to keep all FE stuff cohese

File: package.json [hunk 3]
```diff
@@ -35,6 +44,5 @@
     "ts-node": "9.1.1",
     "typescript": "4.1.3"
   },
-  "types": "./global.d.ts",
```
Comment (clayton-duarte) [inline package.json:38]: We dont need it anymore?

File: package.json (hunk 2 above)
Comment (daviduzumeri) [reply package.json:7]: Yeah I guess we'll never have a non-docs yarn dev, that's fair

File: package.json (hunk 3 above)
Comment (daviduzumeri) [reply package.json:38]: Nope, it's in a types.ts file now

File: src/components/index.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+export { Text }
+export * from './Text'
+
+import TextField from './TextField'
+export { TextField }
+export * from './TextField'
+
+import ThemeProvider from './ThemeProvider'
+export { ThemeProvider }
+export * from './ThemeProvider'
```
Comment (clayton-duarte) [inline src/components/index.ts:85]: really good to have it... I always wondered if there wasn't a way to it in an easier way (like creating this file with a script that looks at the file system and creates all imports/exports), but that's the most human-readable way IMO

File: src/components/types.ts [hunk 5]
```diff
@@ -1,5 +1,9 @@
 import '@emotion/react'
 
+interface Styles {
+  [ruleOrSelector: string]: string | number | Styles
+}
+
```
Comment (clayton-duarte) [inline src/components/types.ts:6]: and it's back, hehehe

Comment (clayton-duarte) [reply src/components/types.ts:6]: BTW, I was the one who created this "recursive" type, but IDK if I'm breaking any rule here

Comment (daviduzumeri) [reply src/components/types.ts:6]: Hahahahahahaa yeah you were right

**Review by clayton-duarte — APPROVED** [review]
Great work man, thank!

File: src/components/index.ts (hunk 4 above)
Comment (daviduzumeri) [reply src/components/index.ts:85]: Yeah this is was annoying to type out lol but it seems that that's how material-ui does it so

**Pull request merged**